### ISSUE TITLE
aregister -P, -C and -U miscalculate initial balance

### ISSUE THREAD

**Issue by hinnerk** [simonmichael/hledger#2214]
Using version hledger 1.33, mac-aarch64 installed with MacPorts the output of `areg` with `--cleared`, `--pending` or `--unmarked` seems identical to `register`. I did not find any existing mentions of this issue (although #1634 looks similar), so I am reporting it without verifying against the latest master branch. I hope this is helpful.

## Minimal example:

```
2024-08-03 * example 1
  expenses:food          $10
  assets:cash

2024-08-03 ! example 2
  expenses:food          $10
  assets:cash

2024-08-03 example 3
  expenses:food          $10
  assets:cash
```

The expected total for each example is $ -30.

## Example for `--pending`

```
$ hledger areg cash --width=65 -P
Transactions in assets:cash and subaccounts:
2024-08-03 example 2     ex:food               $-10          $-10
```

## Example for `--cleared`

```
$ hledger areg cash --width=65 -C
Transactions in assets:cash and subaccounts:
2024-08-03 example 1     ex:food               $-10          $-10
```

## Example for `--unmarked`

```
$ hledger areg cash --width=65 -U
Transactions in assets:cash and subaccounts:
2024-08-03 example 3     ex:food               $-10          $-10
````

**Comment by simonmichael** [comment]
Perhaps it's this: aregister always enables --historical. This just means that generally it will keep showing an accurate historical running balance even if you restrict the start date. But it can't do that if you exclude postings by other criteria (status, above); that will disrupt the running balance.

**Comment by hinnerk** [comment]
TL;DR: I misunderstood the documentation, closing the ticket now.

For reference, I was trying to see how the running balance in an account changes with pending and unmarked future transactions. I was hoping to filter these transactions out and still see the full running balance. It turns out that I misunderstood the documentation regarding `areg` and `reg`. Here's the core of what I understand now:

- `aregister` shows the transactions and running balances in one account, optionally filtered by a query. Use it if you are looking for a traditional bank-like report. For more complex search queries that are fuzzy and cross account boundaries use `register` instead.
- `register` shows postings and their running total, selected by search queries that can be fuzzy and cross account boundaries. If you are looking for a traditional bank-like report over a single account, use `aregister` instead.

I am still working to understand both better, so that I can make sure that my expectations align with the software's behavior.

As it turns out that my bug report was based on my misunderstanding, I'll close it now. I thought you might find it helpful to see how your software is used which is why I included this.

Thank you for building this great software!

**Comment by simonmichael** [comment]
I definitely appreciate the feedback. Your description sounds right to me, though I don't quite get the "fuzzy" - we don't have much of that in hledger.

**Comment by hinnerk** [comment]
I took it from the documentation:

> With -m DESC/--match=DESC, register does a fuzzy search for one recent posting whose description is most similar to DESC.

And wondered how much it would resemble `fzf`.
